dask_kubernetes/core.py

```diff
@@ -511,44 +549,18 @@ def f(to_close):
             except kubernetes.client.rest.ApiException as e:
                 # If a pod has already been removed, just ignore the error
                 if e.status != 404:
-                    raise
+                    raise kubernetes.client.rest.ApiException(e)
```

Comment:
Why this change?  Calling `raise` will raise the exception that we're curently handling.  I think it also retains the original traceback.